src/context.rs

```diff
@@ -1214,7 +1214,7 @@ impl Context {
                 self.sessions.2,
                 &in_data,
                 hashing_algorithm.into(),
-                hierarchy.rh(),
+                hierarchy.esys_rh(),
```

Comment:
The change of handle type for hash wont work. That will break backward compatibility. So unless we change the minimum required version to be latest tpm2-tss 2.4.x version then we cannot change the handle types to be used that easily. That is also why the CI fails. Because you are trying to use a handle that is not supported by the 2.3.3 version that is used for CI and that is the minimum required version.

Reply:
> 
> 
> The change of handle type for hash wont work. That will break backward compatibility. So unless we change the minimum required version to be latest tpm2-tss 2.4.x version then we cannot change the handle types to be used that easily. That is also why the CI fails. Because you are trying to use a handle that is not supported by the 2.3.3 version that is used for CI and that is the minimum required version.

I'm wondering if a solution here would be to have features for the different version "sets", and set one of them by default. I.e. we'd have features like `tss_v2.3.x` and `tss_v2.4.x` and for each one we handle things differently. Would mean more testing, but that's fine by me, as long as we support a broader range of TSS lib versions.

Reply:
Yes, I think that is actually a fantastic idea!
```
tss-esapi = {"4.0.7-alpha.1", features = ["min_compat_tss_v2.4.1"]}
```
Or something like that
 
We just need to be careful that we document all the different available compatibility versions somewhere.